src/main/java/seedu/address/model/recipe/attribute/Ingredient.java

```diff
@@ -3,67 +3,74 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
+import java.util.List;
+
 /**
- * Represents a Recipe's ingredient in its {@code IngredientList}.
- * Guarantees: immutable; is valid as declared in {@link #isValidIngredient(String)}
+ * Represents a Recipe's ingredient in its {@code IngredientList}. Guarantees:
+ * immutable; is valid as declared in {@link #isValidIngredient(String)}
  */
 public class Ingredient {
 
-    public static final String MESSAGE_CONSTRAINTS = "Ingredient can take any values, and it should not be blank";
-
-    /*
-     * The first character of the ingredient must not be a whitespace,
-     * otherwise " " (a blank string) becomes a valid input.
-     */
-    public static final String VALIDATION_REGEX = "[^\\s].*";
+    public static final String MESSAGE_CONSTRAINTS = "Each ingredient should be of the form:"
+            + "'ing/ing1, ing1qty; ing2, ing2qty;...'."
+            + "Spaces are optional. Valid ingredient quantities are specified in the User Guide. ";
 
     public final String name;
     private String quantity;
+    private Quantity quantity2;
 
     /**
-     * Constructs an {@code Ingredient}.
+     * Construct an Ingredient from {@code ingredientString}; the parameter is
+     * required to be non-null, and must follow a pattern.
      *
-     * @param name A valid ingredient string.
+     * @param ingredientString the {@link String} to be decoded
      */
-    public Ingredient(String name, String quantity) {
-        requireNonNull(name);
-        requireNonNull(quantity);
-        checkArgument(isValidIngredient(name), MESSAGE_CONSTRAINTS);
-        checkArgument(isValidIngredient(quantity), MESSAGE_CONSTRAINTS);
-        this.name = name;
-        this.quantity = quantity;
+    public Ingredient(String ingredientString) {
+        requireNonNull(ingredientString);
+
+        List<String> ingredientParts = List.of(ingredientString.split(","));
+        ingredientParts.forEach(String::trim);
+
+        checkArgument(isValidName(ingredientParts.get(0)), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidName(ingredientParts.get(1)), MESSAGE_CONSTRAINTS);
+
+        this.name = ingredientParts.get(0);
+        this.quantity = ingredientParts.get(1);
+        // new Quantity(ingredientParts.get(1));
     }
 
     /**
-     * Returns true if a given string is a valid ingredient.
+     * Returns {@code true} if {@code nameString} is <em>not</em> blank.
+     *
+     * @param nameString the {@link String} to be tested.
      */
-    public static boolean isValidIngredient(String test) {
-        return test.matches(VALIDATION_REGEX);
+    private Boolean isValidName(String nameString) {
+        return !nameString.isBlank();
```

Comment:
This makes way more sense than writing an extra regex just to check that the string is not blank.

Reply:
I think we should standardise how we validate, cuz most of our classes use the earlier method